File: project-euler/problem 10/solution.ts
let prime_numbers: number[] = [];  // list of prime numbers
let number: number = 2;  // starting number

function isPrime(n: number): boolean {
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) return false;
    }
    return true;
}

while (number < 2000000) {
    if (isPrime(number)) {
        prime_numbers.push(number);
    }

    number += 1;
}

console.log(prime_numbers.reduce((partialSum, a) => partialSum + a, 0));

File: project-euler/problem 10/solution.py
import math

prime_numbers: list[int] = []
number = 2


def is_prime(num):
    # Iterates from 2 to sqrt(num)+1 as discussed above
    # Make sure to convert sqrt to int for range
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False

    return True


while number < 2000000:
    if is_prime(number):
        prime_numbers.append(number)

    number += 1


print(sum(prime_numbers))
